numba_cuda/numba/cuda/cudadrv/driver.py

```diff
@@ -2046,32 +1973,51 @@ class ManagedOwnedPointer(OwnedPointer, mviewbuf.MemAlloc):
     pass
 
 
+_DEFAULT_STREAM_REPRS = {
+    0: "<Default CUDA stream>",
+    drvapi.CU_STREAM_DEFAULT: "<Default CUDA stream>",
+    drvapi.CU_STREAM_LEGACY: "<Legacy default CUDA stream>",
+    drvapi.CU_STREAM_PER_THREAD: "<Per-thread default CUDA stream>",
+}
+
+
 class Stream:
-    def __init__(self, handle, finalizer=None, external=False):
```

Comment:
This is a breaking change that will definitely be felt and cause downstream issues. I.E. a quick search on GitHub (https://github.com/search?q=%22cudadrv.driver.Stream%28%22&type=code) shows a small number of projects that use this `__init__` signature.

Should we go through a period of deprecation first?

Reply:
I reverted this change.